src/iperf_error.c

```diff
@@ -113,7 +109,11 @@ iperf_errexit(struct iperf_test *test, const char *format, ...)
 	    cJSON_AddStringToObject(test->json_top, "error", str);
         }
 	iperf_json_finish(test);
-    } else
+    } else {
+        if (pthread_mutex_unlock(&(test->print_mutex)) != 0) {
+            perror("iperf_errexit: pthread_mutex_unlock");
```

Comment:
This code should be locking the mutex, not unlocking it, because we're trying to protect the fprintf() calls in the next few lines.

Reply:
Oops! At least I hope that I mean to lock here.  Changed.